resource/systemd/daemon_reload.go

```diff
@@ -5,6 +5,9 @@ import (
 	"github.com/coreos/go-systemd/dbus"
 )
 
+//CheckDaemonReload checks whether the systemd daemon needs to be reloaded.
+//it does not check if a unit is failed and needs to be reset though.
+//NOTE : should I implement checkResetFailed?
```

Comment:
In Check: 
1. Check if unit is in failed loading state. 
2. Return to the user 

In Apply
1. Check if unit is in a failed loading state.
2. Reset Failed
3. Reload Daemon
4. Wait to Load
5. If unit is no longer failed, continue.